eva-accession-ws/src/test/java/uk/ac/ebi/eva/accession/ws/ClusteredVariantHistoryEndPointTest.java

```diff
@@ -0,0 +1,450 @@
+
+        int fetchHistoryOfRS = 1;
+        String getVariantsUrl = String.format(URL, fetchHistoryOfRS);
+
+        ResponseEntity<VariantHistory<ClusteredVariant, IClusteredVariant, String, Long>> getVariantsResponse =
+                testRestTemplate.exchange(getVariantsUrl, HttpMethod.GET, null, new ClusteredVariantType());
+        assertEquals(HttpStatus.OK, getVariantsResponse.getStatusCode());
+        VariantHistory<ClusteredVariant, IClusteredVariant, String, Long> variantHistory =
+                getVariantsResponse.getBody();
+
+        List<AccessionResponseDTO<ClusteredVariant, IClusteredVariant, String, Long>> allVariants = variantHistory.getVariants();
+        List<HistoryEventDTO<Long, ClusteredVariant>> allOperations = variantHistory.getOperations();
+
+        assertEquals(2, allVariants.size());
+        assertEquals(3, allOperations.size());
+        assertTrue(allVariants.stream().allMatch(v -> v.getAccession().equals(1L)));
+        //Variant with RS 1 present in both ASM1 and ASM2
+        assertTrue(allVariants.stream().allMatch(v -> Arrays.asList("ASM1", "ASM2").contains(v.getData().getAssemblyAccession())));
+        assertTrue(allOperations.stream().allMatch(o -> o.getType().equals(EventType.RS_SPLIT)));
+        // Every split operation from RS 1
+        assertTrue(allOperations.stream().allMatch(o -> o.getAccession().equals(1L)));
+        assertTrue(allOperations.stream().allMatch(o -> Arrays.asList(3000000000L, 3000000001L, 3000000002L).contains(o.getSplitInto())));
+    }
+
+    private void mergeRS(List<SubmittedVariantEntity> submittedVariantEntities) throws Exception {
+        SubmittedVariantOperationEntity mergeOperation = new SubmittedVariantOperationEntity();
+        mergeOperation.fill(RSMergeAndSplitCandidatesReaderConfiguration.MERGE_CANDIDATES_EVENT_TYPE,
+                submittedVariantEntities.get(0).getAccession(), null, "Different RS with matching loci",
+                submittedVariantEntities.stream().map(SubmittedVariantInactiveEntity::new)
+                        .collect(Collectors.toList()));
+
+        rsMergeWriter.write(Collections.singletonList(mergeOperation));
+    }
+
+    private void splitRS(List<SubmittedVariantEntity> submittedVariantEntities) throws Exception {
+        SubmittedVariantOperationEntity splitOperation = new SubmittedVariantOperationEntity();
+        splitOperation.fill(RSMergeAndSplitCandidatesReaderConfiguration.SPLIT_CANDIDATES_EVENT_TYPE,
+                submittedVariantEntities.get(0).getAccession(), "Hash mismatch with " +
+                        submittedVariantEntities.get(0).getClusteredVariantAccession(),
+                submittedVariantEntities.stream().map(SubmittedVariantInactiveEntity::new)
+                        .collect(Collectors.toList())
+        );
+        mongoTemplate.insert(Collections.singletonList(splitOperation), SubmittedVariantOperationEntity.class);
+        rsSplitWriter.write(Collections.singletonList(splitOperation));
```

Comment:
Judgment call: I am not sure explicitly invoking split writer (thereby creating a dependency on the clustering module) is a good idea. We know the end result of splits and merges. In my opinion, we should try to emulate that end result in data rather.